### PR TITLE
docs(styleguide): add deprecation warning and index to pure grids

### LINKED ISSUE #947: Add deprecation warning to Pure grid
Now that we have the flexbox grid/layout component available (http://rackerlabs.github.io/encore-ui/#/component/layout), we need to add a warning to the Pure grid/style guide to indicate that the Pure grid will be deprecated in the near future.

### PR DESCRIPTION
**Pull request by glynnis**
Closes #947

### CODE REVIEW

**Comment by CITguy** [comment]
LGTM

File: demo/styleguide/layouts.html [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <rx-page title="'Layouts'">
 
-    <h3 class="title clear">Grids</h3>
     <div class="component-description clear">
+        <h2>Grids</h2>
+        <p><a href="http://github.com/badges/stability-badges"><img src="http://badges.github.io/stability-badges/dist/deprecated.svg"></a></p>
 
+        <p><strong>NOTE: Use of these grids has been deprecated and this code will be removed in the near future. Any current usage of Pure CSS framework grids should be converted to the new <a href="/component/layout">flexbox grid</a></strong>.
         <p>While EncoreUI was initially designed to be an AngularJS widget library, we recognize that teams will want to use a CSS layout framework. We have added the <a href="http://purecss.io/grids/">Grids module from Yahoo!'s Pure CSS framework</a> for teams to easily build out a grid or column system as part of EncoreUI. We chose this lightweight framework for its emphasis on flex based layouts.</p>
```
Comment (parlarjb) [inline demo/styleguide/layouts.html]: Because we're deprecating, I'd be fine with removing this line completely.

Comment (CITguy) [inline demo/styleguide/layouts.html]: agreed

Comment (glynnis) [inline demo/styleguide/layouts.html]: Updated to remove line about deprecation.

**Comment by parlarjb** [comment]
I think there's been some confusion :) I _liked_ the line you originally added `Use of these grids has been deprecated`. I meant we should get rid of the original `While EncoreUI was initially designed to be an AngularJS widget library, we recognize that teams will want to use a CSS layout framework.` line

**Comment by glynnis** [comment]
Oh, my bad :(  It's always difficult for me to tell the way that github displays line-item comment which line it's referring to. Will re-fix.

**Comment by glynnis** [comment]
@parlarjb fixed

**Pull request merged**